Cython/Compiler/TypeSlots.py

```diff
@@ -1061,7 +1074,8 @@ def __init__(self, old_binops):
             EmptySlot("tp_weaklist"),
             EmptySlot("tp_del"),
             EmptySlot("tp_version_tag"),
-            EmptySlot("tp_finalize", ifdef="PY_VERSION_HEX >= 0x030400a1"),
+            SyntheticSlot("tp_finalize", ["__del__"], "0", ifdef="PY_VERSION_HEX >= 0x030400a1",
+                          used_ifdef="CYTHON_USE_TP_FINALIZE"),
```

Comment:
I can't add the suggestion in the right place, but this also needs:

```
MethodSlot(destructor, "", "__del__", method_name_to_slot)
```
to be added just under
```
MethodSlot(destructor, "", "__dealloc__", method_name_to_slot)
```
for `__del__` to be recognised as a special method.

Reply:
Right, that was originally added in this PR:
https://github.com/cython/cython/blob/d695dce345cf298c4d660c30c2c64a10737c5593/Cython/Compiler/TypeSlots.py#L1010